From: Stanislav Malyshev <stas@php.net>
Date: Mon, 13 Apr 2020 21:00:44 -0700
Subject: Fix bug #79330 - make all execution modes consistent in rejecting \0

(cherry picked from commit 14fcc813948254b84f382ff537247d8a7e5e0e62)
Debian-ELTS: php7.0 7.0.33-0+deb9u26, debian/patches/0126-Fix-bug-79330-make-all-execution-modes-consistent-in.patch
---
 ext/standard/exec.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/ext/standard/exec.c b/ext/standard/exec.c
index d6f0cbfe..e486f52a 100644
--- a/ext/standard/exec.c
+++ b/ext/standard/exec.c
@@ -524,6 +524,15 @@ PHP_FUNCTION(shell_exec)
 		return;
 	}
 
+	if (!command_len) {
+		php_error_docref(NULL, E_WARNING, "Cannot execute a blank command");
+		RETURN_FALSE;
+	}
+	if (strlen(command) != command_len) {
+		php_error_docref(NULL, E_WARNING, "NULL byte detected. Possible attack");
+		RETURN_FALSE;
+	}
+
 #ifdef PHP_WIN32
 	if ((in=VCWD_POPEN(command, "rt"))==NULL) {
 #else
